### PR TITLE
Feature/todo app example

### PR DESCRIPTION
**Pull request by nikolay-vasilev-prime**
#### Overview

> Closes: #734

Adding Todo app to the rx_bloc examples.

#### Checklist

*Implementation*
- [x] Implementation matches ticket acceptance criteria and technical notes
- [x] Manually tested against Acceptance Criteria
- [x] UI checked in Light / Dark mode

*Stability*
- [x] Checked if changes affect any features and verified affected features work as expected
- [x] Added unit tests for new code and verified existing tests work as expected

*Code quality*
- [x] Updated `CHANGELOG.md`, `README.md` and package versions in `pubspec.yaml`
- [x] Dependencies are updated to latest versions or new tickets are created if there are breaking changes or deprecations
- [x] If an unrelated part of the codebase needs to be updated or refactored, create tickets with proposed changes

### CODE REVIEW

File: examples/todoapp/lib/base/data_sources/remote/deep_link_remote_data_source.dart [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+@RestApi()
+abstract class DeepLinkRemoteDataSource {
+  factory DeepLinkRemoteDataSource(Dio dio, {String baseUrl}) =
+      _DeepLinkRemoteDataSource;
+
+  @GET('/api/deep-links')
+  Future<DeepLinkListResponseModel> fetchDeepLinkList();
+
+  @GET('/api/deep-links/{id}')
```
Comment (StanevPrime) [inline examples/todoapp/lib/base/data_sources/remote/deep_link_remote_data_source.dart:24]: As agreed all generated by the rx_bloc_cli files, which are not used need to be removed

File: examples/todoapp/lib/base/data_sources/remote/push_notification_data_source.dart [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import 'package:dio/dio.dart';
+import 'package:retrofit/http.dart';
+
+import '../../models/request_models/push_message_request_model.dart';
+import '../../models/request_models/push_notification_data_request_model.dart';
+
+part 'push_notification_data_source.g.dart';
+
+@RestApi()
+abstract class PushNotificationsDataSource {
```
Comment (StanevPrime) [inline examples/todoapp/lib/base/data_sources/remote/push_notification_data_source.dart:17]: As agreed all generated by the rx_bloc_cli files, which are not used need to be removed

File: examples/todoapp/lib/base/models/deep_link_model.dart [hunk 3]
```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2023, Prime Holding JSC
+// https://www.primeholding.com
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:json_annotation/json_annotation.dart';
```
Comment (StanevPrime) [inline examples/todoapp/lib/base/models/deep_link_model.dart:8]: As agreed all generated by the rx_bloc_cli files, which are not used need to be removed

File: examples/todoapp/lib/base/models/request_models/push_message_request_model.dart [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:json_annotation/json_annotation.dart';
+
+part 'push_message_request_model.g.dart';
+
+@JsonSerializable()
+class PushMessageRequestModel {
```
Comment (StanevPrime) [inline examples/todoapp/lib/base/models/request_models/push_message_request_model.dart:13]: As agreed all generated by the rx_bloc_cli files, which are not used need to be removed

File: examples/todoapp/lib/base/models/request_models/push_notification_data_request_model.dart [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:json_annotation/json_annotation.dart';
+
+part 'push_notification_data_request_model.g.dart';
+
+@JsonSerializable()
+class PushNotificationDataRequestModel {
```
Comment (StanevPrime) [inline examples/todoapp/lib/base/models/request_models/push_notification_data_request_model.dart:13]: As agreed all generated by the rx_bloc_cli files, which are not used need to be removed

File: examples/todoapp/lib/base/models/response_models/deep_link_list_response_model.dart [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+// https://opensource.org/licenses/MIT.
+
+import 'package:json_annotation/json_annotation.dart';
+
+import '../deep_link_model.dart';
+
+part 'deep_link_list_response_model.g.dart';
+
+@JsonSerializable()
+class DeepLinkListResponseModel {
```
Comment (StanevPrime) [inline examples/todoapp/lib/base/models/response_models/deep_link_list_response_model.dart:15]: As agreed all generated by the rx_bloc_cli files, which are not used need to be removed

File: examples/todoapp/lib/base/repositories/deep_link_repository.dart [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import '../common_mappers/error_mappers/error_mapper.dart';
+import '../data_sources/remote/deep_link_remote_data_source.dart';
+import '../models/deep_link_model.dart';
+import '../models/response_models/deep_link_list_response_model.dart';
+
+class DeepLinkRepository {
```
Comment (StanevPrime) [inline examples/todoapp/lib/base/repositories/deep_link_repository.dart:13]: As agreed all generated by the rx_bloc_cli files, which are not used need to be removed

File: examples/todoapp/lib/l10n/arb/feature_deep_link/en.arb [hunk 8]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (StanevPrime) [inline examples/todoapp/lib/l10n/arb/feature_deep_link/en.arb:1]: Please remove all files related to not-used features

File: examples/todoapp/lib/l10n/arb/feature_deep_link/bg.arb
Comment (StanevPrime) [inline examples/todoapp/lib/l10n/arb/feature_deep_link/bg.arb:1]: Please remove all files related to not-used features

File: examples/todoapp/lib/l10n/arb/feature_enter_message/bg.arb
Comment (StanevPrime) [inline examples/todoapp/lib/l10n/arb/feature_enter_message/bg.arb:1]: Please remove all files related to not-used features

File: examples/todoapp/lib/l10n/arb/feature_enter_message/en.arb
Comment (StanevPrime) [inline examples/todoapp/lib/l10n/arb/feature_enter_message/en.arb:1]: Please remove all files related to not-used features

File: examples/todoapp/lib/l10n/arb/feature_notifications/bg.arb
Comment (StanevPrime) [inline examples/todoapp/lib/l10n/arb/feature_notifications/bg.arb:1]: Please remove all files related to not-used features

File: examples/todoapp/lib/l10n/arb/feature_notifications/en.arb
Comment (StanevPrime) [inline examples/todoapp/lib/l10n/arb/feature_notifications/en.arb:1]: Please remove all files related to not-used features

File: examples/todoapp/lib/l10n/arb/feature_profile/bg.arb
Comment (StanevPrime) [inline examples/todoapp/lib/l10n/arb/feature_profile/bg.arb:1]: Please remove all files related to not-used features

File: examples/todoapp/lib/l10n/arb/feature_profile/en.arb
Comment (StanevPrime) [inline examples/todoapp/lib/l10n/arb/feature_profile/en.arb:1]: Please remove all files related to not-used features

File: examples/todoapp/lib/l10n/network_lookup/deep_link_lookup.dart
Comment (StanevPrime) [inline examples/todoapp/lib/l10n/network_lookup/deep_link_lookup.dart:1]: Please remove all files related to not-used features

File: examples/todoapp/lib/l10n/network_lookup/enter_message_lookup.dart
Comment (StanevPrime) [inline examples/todoapp/lib/l10n/network_lookup/enter_message_lookup.dart:1]: Please remove all files related to not-used features

File: examples/todoapp/lib/lib_router/routes/showcase_routes.dart
Comment (StanevPrime) [inline examples/todoapp/lib/lib_router/routes/showcase_routes.dart:1]: Please remove all files related to not-used features

**Review by StanevPrime — CHANGES_REQUESTED** [review]

File: examples/todoapp/pubspec.yaml [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+        - asset: assets/fonts/WorkSans/WorkSans-Regular.ttf
+          weight: 400
+        - asset: assets/fonts/WorkSans/WorkSans-Medium.ttf
+          weight: 500
+
+r_flutter:
+  intl: lib/l10n/arb/en.arb
+  intl_features:
+    - name: error
+    - name: feature_deep_link
```
Comment (StanevPrime) [inline examples/todoapp/pubspec.yaml:78]: Please remove all configurations related to not-used features

**Review by StanevPrime — CHANGES_REQUESTED** [review]

File: examples/todoapp/lib/base/data_sources/remote/deep_link_remote_data_source.dart (hunk 1 above)
Comment (nikolay-vasilev-prime) [reply examples/todoapp/lib/base/data_sources/remote/deep_link_remote_data_source.dart]: Done

File: examples/todoapp/lib/base/data_sources/remote/push_notification_data_source.dart (hunk 2 above)
Comment (nikolay-vasilev-prime) [reply examples/todoapp/lib/base/data_sources/remote/push_notification_data_source.dart]: Done

File: examples/todoapp/lib/base/models/deep_link_model.dart (hunk 3 above)
Comment (nikolay-vasilev-prime) [reply examples/todoapp/lib/base/models/deep_link_model.dart]: Done

File: examples/todoapp/lib/base/models/request_models/push_message_request_model.dart (hunk 4 above)
Comment (nikolay-vasilev-prime) [reply examples/todoapp/lib/base/models/request_models/push_message_request_model.dart]: Done

File: examples/todoapp/lib/base/models/request_models/push_notification_data_request_model.dart (hunk 5 above)
Comment (nikolay-vasilev-prime) [reply examples/todoapp/lib/base/models/request_models/push_notification_data_request_model.dart]: Done

File: examples/todoapp/lib/base/models/response_models/deep_link_list_response_model.dart (hunk 6 above)
Comment (nikolay-vasilev-prime) [reply examples/todoapp/lib/base/models/response_models/deep_link_list_response_model.dart]: Done

File: examples/todoapp/lib/base/repositories/deep_link_repository.dart (hunk 7 above)
Comment (nikolay-vasilev-prime) [reply examples/todoapp/lib/base/repositories/deep_link_repository.dart]: Done

File: examples/todoapp/lib/l10n/arb/feature_deep_link/en.arb (hunk 8 above)
Comment (nikolay-vasilev-prime) [reply examples/todoapp/lib/l10n/arb/feature_deep_link/en.arb]: Done

File: examples/todoapp/lib/l10n/arb/feature_deep_link/bg.arb
Comment (nikolay-vasilev-prime) [reply examples/todoapp/lib/l10n/arb/feature_deep_link/bg.arb:1]: Done

File: examples/todoapp/lib/l10n/arb/feature_enter_message/bg.arb
Comment (nikolay-vasilev-prime) [reply examples/todoapp/lib/l10n/arb/feature_enter_message/bg.arb:1]: Done

File: examples/todoapp/lib/l10n/arb/feature_enter_message/en.arb
Comment (nikolay-vasilev-prime) [reply examples/todoapp/lib/l10n/arb/feature_enter_message/en.arb:1]: Done

File: examples/todoapp/lib/l10n/arb/feature_notifications/bg.arb
Comment (nikolay-vasilev-prime) [reply examples/todoapp/lib/l10n/arb/feature_notifications/bg.arb:1]: Done

File: examples/todoapp/lib/l10n/arb/feature_notifications/en.arb
Comment (nikolay-vasilev-prime) [reply examples/todoapp/lib/l10n/arb/feature_notifications/en.arb:1]: Done

File: examples/todoapp/lib/l10n/arb/feature_profile/bg.arb
Comment (nikolay-vasilev-prime) [reply examples/todoapp/lib/l10n/arb/feature_profile/bg.arb:1]: Done

File: examples/todoapp/lib/l10n/arb/feature_profile/en.arb
Comment (nikolay-vasilev-prime) [reply examples/todoapp/lib/l10n/arb/feature_profile/en.arb:1]: Done

File: examples/todoapp/lib/l10n/network_lookup/deep_link_lookup.dart
Comment (nikolay-vasilev-prime) [reply examples/todoapp/lib/l10n/network_lookup/deep_link_lookup.dart:1]: Done

File: examples/todoapp/lib/l10n/network_lookup/enter_message_lookup.dart
Comment (nikolay-vasilev-prime) [reply examples/todoapp/lib/l10n/network_lookup/enter_message_lookup.dart:1]: Done

File: examples/todoapp/lib/lib_router/routes/showcase_routes.dart
Comment (nikolay-vasilev-prime) [reply examples/todoapp/lib/lib_router/routes/showcase_routes.dart:1]: Done

File: examples/todoapp/pubspec.yaml (hunk 9 above)
Comment (nikolay-vasilev-prime) [reply examples/todoapp/pubspec.yaml:78]: Done

**Review by StanevPrime — APPROVED** [review]

**Pull request merged**